src/CPUSolver.cpp

```diff
@@ -663,24 +665,22 @@ void CPUSolver::transferBoundaryFlux(int track_id,
   int bc;
   int track_out_id;
 
-  /* Extract boundary conditions for this Track */
-
   /* For the "forward" direction */
   if (direction) {
-    start = _tracks[track_id]->isReflOut() * _polar_times_groups;
-    bc = (int)_tracks[track_id]->getBCOut();
+    start = _tracks[track_id]->isNextOut() * _polar_times_groups;
+    bc = std::min((int)_tracks[track_id]->getBCOut(), 1);
     track_out_id = _tracks[track_id]->getTrackOut()->getUid();
   }
```

Comment:
Not for this PR, but in the future I would like to see an abstraction for direction